### DISCUSSION TITLE
Process Interface evaluation

### DISCUSSION THREAD

**Discussion by schettino72** [BAMresearch/NFDI4IngScientificWorkflowRequirements#57] [General]
I think this is a great criteria that makes a lot of difference in the "easiness" of creating certain workflows.
But the evaluations options has too much focus on being well defined...

1.  The workflow system is purely file-based and does not define interface formats.

2.  The workflow system has a file and non-file based interface, where the non-file based inputs are well defined.

3.  The workflow system has a file and non-file based interface, where both the file and non-file based inputs are well defined.

`doit` is being classified as *1*, that is not correct. 
Although it could also not be classified as *2* because the interface is not well-defined.

I guess you should add another level or drop the "well defined" for *2*.

**Comment by joergfunger** [comment]
The focus of that requirement was that (in particular when multiple users are working on some parts of a workflow) it is advantageous to define interfaces (that in theory could even be checked a priori. This is often tricky for files since one could certainly define some structure and then test that, but there was no tool providing that currently. As for non-file parameters, a couple of tools provide a direct definition of the type. which makes it easier to jointly work on this since one can clearly define the interfaces (and even get errors if a task does return something else). In that sense, the focus of this requirement was really the interface definition. On the other hand, I agree that the information of allowing file and non-file based parameters being passed is another criterium. We will discuss that in our next meeting to evaluate, if we should introduce another criterium or modify the current one. Thanks a lot for your valuable input.
